### PR TITLE
MODFQMMGR-116 Add valueFunctions

### PR DESCRIPTION
**Pull request by mweaver-ebsco**

### CODE REVIEW

File: src/main/resources/db/changelog/changes/v1.1.0/update-drv-purchase-order-line-details-definition.xml [hunk 1]
```diff
@@ -21,6 +21,7 @@
               },
               "valueGetter": "( SELECT array_agg(acq_id.value::text) FILTER (WHERE (acq_id.value::text) IS NOT NULL) AS array_agg FROM jsonb_array_elements_text(purchase_order.jsonb -> 'acqUnitIds'::text) acq_id(value))",
               "filterValueGetter": "( SELECT array_agg(lower(acq_id.value::text)) FILTER (WHERE (acq_id.value::text) IS NOT NULL) AS array_agg FROM jsonb_array_elements_text(purchase_order.jsonb -> 'acqUnitIds'::text) acq_id(value))",
+              "valueFunction": "lower(:value)",
```
Comment (mweaver-ebsco) [inline src/main/resources/db/changelog/changes/v1.1.0/update-drv-purchase-order-line-details-definition.xml:24]: @bvsharp, these valueGetters are pretty intense, but it just looks like we need to make sure the value is lower-case, in order to match. Does that sound right to you?

File: src/main/resources/db/changelog/changes/v1.1.0/update-drv-loan-details-definition.xml [hunk 2]
```diff
@@ -69,6 +70,7 @@
                 "dataType": "stringType"
               },
               "filterValueGetter": "lower(item_details.jsonb ->> 'barcode'::text)",
+              "valueFilter": "lower(:value)",
```
Comment (mweaver-ebsco) [inline src/main/resources/db/changelog/changes/v1.1.0/update-drv-loan-details-definition.xml:73]:
```suggestion
              "valueFunction": "lower(:value)",
```

File: src/main/resources/db/changelog/changes/v1.1.0/update-drv-loan-details-definition.xml [hunk 3]
```diff
@@ -100,6 +103,7 @@
               },
               "idColumnName": "item_material_type_id",
               "filterValueGetter": "lower(${tenant_id}_mod_inventory_storage.f_unaccent(material_type_ref_data.jsonb ->> 'name'::text))",
+              "valueFilter": "lower(${tenant_id}_mod_inventory_storage.f_unaccent(:value))",
```
Comment (mweaver-ebsco) [inline src/main/resources/db/changelog/changes/v1.1.0/update-drv-loan-details-definition.xml:106]:
```suggestion
              "valueFunction": "lower(${tenant_id}_mod_inventory_storage.f_unaccent(:value))",
```

File: src/main/resources/db/changelog/changes/v1.1.0/update-drv-item-details-definition.xml [hunk 4]
```diff
@@ -74,6 +75,7 @@
                 "dataType": "dateType"
               },
               "filterValueGetter": "\"left\"(lower((instance_details.jsonb -> 'metadata'::text) ->> 'updatedDate'::text), 600)",
+              "valueFunction": "\"left\"(lower((:value) ->> 'updatedDate'::text), 600)",
```
Comment (mweaver-ebsco) [inline src/main/resources/db/changelog/changes/v1.1.0/update-drv-item-details-definition.xml:78]:
```suggestion
              "valueFunction": "\"left\"(lower(:value), 600)",
```

File: src/main/resources/db/changelog/changes/v1.1.0/update-drv-item-details-definition.xml [hunk 5]
```diff
@@ -284,6 +293,7 @@
                 "dataType": "stringType"
               },
               "filterValueGetter": "\"left\"(lower(${tenant_id}_mod_inventory_storage.f_unaccent((src_inventory_item.jsonb -> 'status'::text) ->> 'name'::text)), 600)",
+              "valueFunction": "\"left\"(lower(${tenant_id}_mod_inventory_storage.f_unaccent((:value) ->> 'name'::text)), 600)",
```
Comment (mweaver-ebsco) [inline src/main/resources/db/changelog/changes/v1.1.0/update-drv-item-details-definition.xml:296]:
```suggestion
              "valueFunction": "\"left\"(lower(${tenant_id}_mod_inventory_storage.f_unaccent(:value)), 600)",
```

File: src/main/resources/db/changelog/changes/v1.1.0/update-drv-loan-details-definition.xml [hunk 6]
```diff
@@ -78,6 +80,7 @@
                 "dataType": "stringType"
               },
               "filterValueGetter": "\"left\"(lower((item_details.jsonb -> 'effectiveCallNumberComponents'::text) ->> 'callNumber'::text), 600)",
+              "valueFunction": "\"left\"(lower((:value) ->> 'callNumber'::text), 600)",
```
Comment (mweaver-ebsco) [inline src/main/resources/db/changelog/changes/v1.1.0/update-drv-loan-details-definition.xml:83]:
```suggestion
              "valueFunction": "\"left\"(lower(:value), 600)",
```

**Review by ncovercash — APPROVED** [review]

**Review by kjain110 — APPROVED** [review]

File: src/main/resources/db/changelog/changes/v1.1.0/update-drv-purchase-order-line-details-definition.xml (hunk 1 above)
Comment (bvsharp) [reply src/main/resources/db/changelog/changes/v1.1.0/update-drv-purchase-order-line-details-definition.xml:24]: yeah, that's right

File: src/main/resources/db/changelog/changes/v1.1.0/update-drv-item-details-definition.xml [hunk 7]
```diff
@@ -271,6 +279,7 @@
               },
               "idColumnName": "item_permanent_location_id",
               "filterValueGetter": "lower(${tenant_id}g_mod_inventory_storage.f_unaccent(permanent_location_ref_data.jsonb ->> 'name'::text))",
```
Comment (bvsharp) [inline src/main/resources/db/changelog/changes/v1.1.0/update-drv-item-details-definition.xml:281]:
```suggestion
              "filterValueGetter": "lower(${tenant_id}_mod_inventory_storage.f_unaccent(permanent_location_ref_data.jsonb ->> 'name'::text))",
```

**Pull request merged**